### PR TITLE
Implement DKG signing

### PR DESCRIPTION
**Pull request by temld4**
refactor MultiPartyECDSASettings for multiple stages
add offline signing stage

### CODE REVIEW

File: beefy-gadget/src/dkg.rs [hunk 1]
```diff
@@ -53,29 +60,194 @@ pub trait MultiPartyECDSAKeyStore: SyncCryptoStore {
...
+				}
+			}
+			_ => {
+				warn!("Signature is not ready");
+				return None;
+			}
+		}
+	}
+
+	pub fn is_ready_to_sing(&self) -> bool {
```
Comment (drewstone) [inline beefy-gadget/src/dkg.rs]:
```suggestion
	pub fn is_ready_to_sign(&self) -> bool {
```

File: beefy-gadget/src/dkg.rs [hunk 2]
```diff
@@ -160,24 +409,259 @@ impl MultiPartyECDSASettings {
...
+			for party in &mut parties.into_iter() {
+				party.proceed();
+
+				if let Some(mut msgs) = party.get_outgoing_messages() {
+					msgs_pull.append(&mut msgs);
+				}
+			}
+
+			for party in &mut parties.into_iter() {
+				println!("Trying sing for party {}, Stage: {:?}", party.party_index, party.stage);
```
Comment (drewstone) [inline beefy-gadget/src/dkg.rs]:
```suggestion
				println!("Trying sign for party {}, Stage: {:?}", party.party_index, party.stage);
```

**Review by drewstone — COMMENTED** [review]
Does this hook up `handle_incoming` messages from the gossip network?

**Comment by temld4** [comment]
>>Does this hook up handle_incoming messages from the gossip network?

It does, but worker requires some modification. I'll commit the changes to this PR.

**Pull request merged**